Modules/Devices/src/Devices.Application/Clients/Commands/UpdateClient/Handler.cs

```diff
@@ -18,23 +18,17 @@ public Handler(IOAuthClientsRepository oAuthClientsRepository, ITiersRepository
 
     public async Task<UpdateClientResponse> Handle(UpdateClientCommand request, CancellationToken cancellationToken)
     {
-        var client = await _oAuthClientsRepository.Find(request.ClientId, cancellationToken);
-
-        if (!string.IsNullOrEmpty(request.TierId))
-        {
-            var tierIdResult = TierId.Create(request.TierId);
-            if (tierIdResult.IsFailure)
-                throw new ApplicationException(ApplicationErrors.Devices.InvalidTierId());
-
-            _ = await _tiersRepository.FindById(tierIdResult.Value, cancellationToken) ?? throw new ApplicationException(GenericApplicationErrors.NotFound(nameof(Tier)));
-        }
-        else
-        {
-            var basicTier = await _tiersRepository.GetBasicTierAsync(cancellationToken);
-            request.TierId = basicTier.Id.Value;
-        }
-
-        client.TierId = request.TierId;
+        var client = await _oAuthClientsRepository.Find(request.ClientId, cancellationToken, track: true);
+
+        var tierIdResult = TierId.Create(request.DefaultTier);
+        if (tierIdResult.IsFailure)
+            throw new ApplicationException(ApplicationErrors.Devices.InvalidTierIdOrDoesNotExist());
+
+        var tierExists = await _tiersRepository.ExistsWithId(tierIdResult.Value, cancellationToken);
+        if (!tierExists)
+            throw new ApplicationException(ApplicationErrors.Devices.InvalidTierIdOrDoesNotExist());
+
+        client.DefaultTier = request.DefaultTier;
```

Comment:
Once again: never add public setters to domain classes.

```suggestion
        client.ChangeDefaultTier(request.DefaultTier);
```